apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/file/InputObjectBuilder.kt

```diff
@@ -42,5 +48,35 @@ internal class InputObjectBuilder(
           .makeDataClassFromParameters(fields.map {
             it.toNamedType().toParameterSpec(context)
           })
+          .addBuilder()
           .build()
+
+  private fun TypeSpec.Builder.addBuilder(): TypeSpec.Builder {
+    if (inputObject.fields.isEmpty()) {
+      return this
```

Comment:
For operations, you're adding a default empty `Builder` while here (input objects), you're not adding any builder at all. I don't mind either solution but the choice should be consistent

Reply:
I agree that's weird, but that's ported straight from Apollo Android. IMHO there should always be a builder there, so I'll see about changing that.

Reply:
Make sense 👍 . Agree on `Builder` all the times, let's make all the things consistent.

Reply:
I made this change, and now I know why this was never noticed - there is no test for it. 🤦  I can sort of see why - an empty input type is unusual.

Reply:
I don't think it's legal to have an input with no fields:
<img width="529" alt="image" src="https://user-images.githubusercontent.com/746336/187303631-dd381a02-7003-446d-8948-77d910409e25.png">
So maybe this is an edge case not worth worrying about too much.

Reply:
Good catch. As it turns out, it's completely invalid to have an input object with no input field. See [the spec](https://spec.graphql.org/draft/#sec-Input-Objects.Type-Validation):

```
An Input Object type must define one or more input fields.
```

So nevermind my initial comment. Input objects with no input fields should never happen. Maybe just add a comment in the source, make sure we don't fall into that trap again?

Reply:
Interesting. Yeah I'll do that.

Reply:
Okay I reverted my change and added a comment.